### DISCUSSION TITLE
Backup Script Error:  .env file is missing

### DISCUSSION THREAD

**Discussion by f1nch87** [solectrus/solectrus#3697] [Q&A]
Hey, 
just wanted to finally redo my Docker setup because i initially build my container with portainer. Now im more familiar with the docker cli itself and wanna do it there.
After long searching i found the solectrus folder with the docker-compose.yml and the .env file.
When i try to execute the script it tells me that there is no .env file. But in the folder there is the .env file of the solectrus stack.
Can anyone help maybe?

```
root@pSolectrus:/var/lib/docker/volumes/portainer_data/_data/compose/1# ls
backup.sh  docker-compose.yml  restore.sh  stack.env
root@pSolectrus:/var/lib/docker/volumes/portainer_data/_data/compose/1# ./backup.sh
Error: .env file is missing. Please ensure the .env file is present.
root@pSolectrus:/var/lib/docker/volumes/portainer_data/_data/compose/1#
```

**Comment by gereons** [comment]
Ist evtl. `stack.env` die gesuchte Datei? (Hab keine Ahnung von portainer aber Google sagt dass da wohl hin und wieder dieser Name verwendet wird...)

In dem Fall würde ein `ln stack.env .env` helfen, dann ist die Datei unter beiden Namen erreichbar.

**Reply by f1nch87** [reply]
stack.env ist die gesuchte Datei und die ist ja auch dort im Ordner enthalten und enthält die solectrus enviroment variablen.
Ich verstehe es nicht so richtig, weil wenn ich das script richtig lese, dann sucht es einfach nach ein .env datei in dem Ordner und die ist ja da.

**Reply by f1nch87** [reply]
ok mit dem symlink hat es funktioniert. Zumindest mit kommt er jetzt weiter zum nächsten Fehler :D.
Jetzt findet er meine Postgressql Container nicht.
Ich glaube ich suche mir einfach alle ordner zusammen und erstelle den container mit copy und paste der dateien direkt.

Und gelernt habe ich kein Portainer mehr zu benutzen :D

**Reply by f1nch87** [reply]
Würde /var/lib/influxdb2 und /var/lib/postgressql/data ausreichen um alle historischen daten wieder zu haben?

**Reply by gereons** [reply]
> dann sucht es einfach nach ein .env datei

Nein, gesucht wird nicht "eine Datei mit .env hinten" sondern eine Datei die genau nur `.env` heisst.

> Jetzt findet er meine Postgressql Container nicht.

Das Script erwartet dass der Container "db" oder "postgresql" heisst. Die Namen bei dir sagt dir `docker ps`, notfalls einfach das Script editieren (Zeile 33, `grep -E '^(postgresql|db)$` könntest du in `grep -E '^DEINCONTAINERNAME$` ändern).

> Würde /var/lib/influxdb2 und /var/lib/postgressql/data ausreichen um alle historischen daten wieder zu haben?  

Ja, solange du die gleiche `compose.yml` und `.env` (umbenannte bisherige `stack.env`) weiterhin benutzt.

**Reply by f1nch87** [reply]
Ok danke.
Ich benenne mal die Container um und arbeite mich durchs script;).

Aber auf jedenfall schonmal schön das es das script gibt.

**Reply by ledermann** [reply]
Das Backup/Restore-Script wurde nicht mit Blick auf Portainer erstellt. Ich weiß auch nicht, ob das so ideal wäre, weil die Grundidee von Portainer ein browserbasiertes UI ist, zu dem ein per SSH zu startendes Bash-Script nicht so gut passt.

Da Portainer aber auch existierende Container anzeigt (also die nicht mit Portainer erstellt wurden), ist vielleicht der umgekehrte Weg sinnvoll: SOLECTRUS auf herkömmliche Weise installieren (in einem Ordner der Wahl, mit `.env` und `compose.yaml`) und die Backup/Restore-Scripte dazu packen. Portainer wird dann die Container anzeigen - wenngleich dann keine Änderungen zugelassen werden.

**Reply by f1nch87** [reply]
Ja dann brauche ich kein Portainer denn dann kann ich nichts bearbeiten und muss trotzdem immer wieder auf die shell.
Und Portainer zu installieren um zu sehen das alles läuft is irgendwie nutzlos. Das habe ich zumindest für mich gelernt.
Ich hab jetzt das postgresql und influxdb volume gefunden und kopiere es sauber auf die "neue" docker vm, packe alles was der container braucht in einen ordner und deploye es normal mit docker und verwalte es mit dockge.

**Comment by f1nch87** [comment]
So, Container sauber neu aufgesetzt mit ner ordentlichen Ordnerstruktur und das Backup Script funktioniert einwandfrei.

Danke dafür
